kubermatic-addons/custom-addon/flux2/01-bootstrap.yaml

```diff
@@ -0,0 +1,45 @@
+{{- $httpCredentialsProvided := and (not (empty .Variables.gitRepositoryHttpUsername)) (not (empty .Variables.gitRepositoryHttpPassword)) -}}
+{{- $authenticationRequired := $httpCredentialsProvided -}}
+---
+{{- if $authenticationRequired }}
+apiVersion: v1
+kind: Secret
+metadata:
+  name: cluster-sync-credentials
+  namespace: flux-system
+data:
+  {{- if $httpCredentialsProvided }}
+  password: {{ .Variables.gitRepositoryHttpUsername | b64enc | quote }}
+  username: {{ .Variables.gitRepositoryHttpPassword | b64enc | quote }}
+  {{- end }}
+{{- end }}
+---
+apiVersion: source.toolkit.fluxcd.io/v1beta1
+kind: GitRepository
+metadata:
+  name: cluster-sync
+  namespace: flux-system
+spec:
+  interval: {{ default "1m" .Variables.updateInterval | quote }}
+  {{- with .Variables.gitRepositoryBranch }}
+  ref:
+    branch: {{ . | quote }}
+  {{- end }}
+  {{- if $authenticationRequired }}
+  secretRef:
+    name: cluster-sync-credentials
+  {{- end }}
+  url: {{ .Variables.gitRepositoryUrl | quote }}
+---
+apiVersion: kustomize.toolkit.fluxcd.io/v1beta2
+kind: Kustomization
+metadata:
+  name: cluster-sync
+  namespace: flux-system
+spec:
+  interval: {{ default "1m" .Variables.updateInterval | quote }}
+  path: {{ default (printf "./clusters/%s" .Cluster.Name) .Variables.kustomizationPath | quote }}
```

Comment:
`.Cluster.Name` is the ID for sure? isn't there `.Cluster.Id`?

Reply:
@michalvanco yes, the [ClusterData.Name](https://github.com/kubermatic/kubermatic/blob/v2.18.2/pkg/addon/template.go#L160) contains the auto-generated, internal cluster name and there is no dedicated field `Id`.